### DISCUSSION TITLE
Any plan for multiple platform support ?

### DISCUSSION THREAD

**Discussion by mortal-dog** [rwf2/Rocket#1907] [Ideas]
I kind of hope Rocket can be used to build Windows MacOS Linux apps. Could it be possible ?

**Comment by memoryruins** [comment]
It should already be possible. Have you experienced any issues while targeting these platforms?

**Reply by mortal-dog** [reply]
WOW, COOL !, I'm totally new to Rocket. I just read the description of Rocket which is "Rocket is an async web framework for Rust with a focus on usability, security, extensibility, and speed." And I thought Rocket is only for web app development. I will try to target other platforms. 

Thanks very much

**Comment by jebrosen** [comment]
> I kind of hope Rocket can be used to build Windows MacOS Linux apps. Could it be possible ?

Not by itself, if I understand the question right. Rocket is a web server framework, so its external input/output is through the HTTP protocol. In principle, it is possible to use Rocket to implement a backend server for just about any kind of program imaginable on Windows, macOS, or Linux. However, it does not include anything in the way of graphics, user interface, or similar functionality; for that, you would need to either write a native or cross-platform GUI application or use the web platform (HTML, CSS, JavaScript) to implement the corresponding interface.

There is still some variety in these approaches: the backend web server could only provide data for a separate GUI to display, or the backend could create an entire application through server-side rendering of HTML templates. Since it's HTTP, the backend could run on the same or a different computer from the user interface. But either way, a web client of some sort (e.g. a web browser, or a GUI that makes API calls over HTTP) would need to be part of the system; if you need a GUI and have no or minimal need for any kind of backend, Rocket or any other web server may not be suited for the overall project.
